gap_linux.go

```diff
@@ -125,6 +125,26 @@ func (a *Advertisement) Stop() error {
 	return nil
 }
 
+// SetServiceData sets the service data for the advertisement.
+func (a *Advertisement) SetServiceData(sd []ServiceDataElement) error {
```

Comment:
@aykevl do you happen to know how I can do this? :smile_cat:

Reply:
What is your goal?

If you want to change advertisement data then I think you can stop and restart the advertisement with the new service data. Looking at the API I don't see a straightforward way to update any part of the advertisement. But this is straightforward to do from userspace: simply stop and restart the advertisement with the new parameters.

See:
https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/org.bluez.LEAdvertisement.rst
https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/org.bluez.LEAdvertisingManager.rst

Reply:
> If you want to change advertisement data then I think you can stop and restart the advertisement 

OK I will probably do that in a separate PR.